src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -363,12 +371,13 @@ using MergeRegexPatternAndFlagsExpression =
     StringExpressionImpl<2, decltype(mergeFlagsIntoRegex), LiteralFromIdGetter>;
 
 [[maybe_unused]] auto replaceImpl =
-    [](std::optional<std::string> input, const std::shared_ptr<RE2>& pattern,
+    [](std::optional<ad_utility::triple_component::Literal> s,
+       const std::unique_ptr<re2::RE2>& pattern,
```

Comment:
during your merge, you changed `shared_ptr` to `unique_ptr` which caused the really ugly template errors. Make the regex argument a shared_ptr again and you will get to more normal errors.